bin/upgrading/update-db-general.pl

```diff
@@ -3120,6 +3120,31 @@
 )
 EOC
 
+register_tablecreate( "profile_services", <<'EOC');
+CREATE TABLE profile_services (
+    name VARCHAR(40) NOT NULL,
```

Comment:
we should use an index as the primary key so that services can rename themselves. with the table design as-is, we can't rename a service.

the traditional method here is to use an `auto_increment` id column, then you don't really have to think about it, look at `priv_list` as an example.

your keys would then be a primary key on profile_service_id and a unique key on name (since presumably we want to enforce uniqueness, but also allow to lookup by name)

```sql
create table profile_services (
    profile_service_id mediumint(8) unsigned NOT NULL auto_increment,
    ...

    primary key (profile_service_id),
    unique key (name)
)```

you would then change the `user_profile_accts` table to contain the profile_service_id instead of the name

Reply:
this is what I get for doing a top down review and commenting. I see you already considered this and are putting the ML string into the db in a separate column. so, the "need" for using an integer is less and makes this more of a nit. I guess it's okay.

Reply:
Yeah, I was following the pattern of the userprop names being fixed forever, but I see your point.